### PR TITLE
FIX: Don't send image sizes for emojis/avatars

### PR DESCRIPTION
**Pull request by pmusaraj**
#### What's the problem? 

Related: https://meta.discourse.org/t/review-media-unless-trust-level-setting-causes-reviews-with-emojis/230189 

When using the "review media unless trust level" setting, posts with emojis or quotes will end up in the review queue even though they don't have any uploaded media. That is because our heuristic for this in the new post manager relies on `image_sizes`: 

https://github.com/discourse/discourse/blob/main/lib/new_post_manager.rb#L117-L118 

Those image sizes are sent from the frontend, and they are gathered via the preview pane. Currently, the selector includes all images, uploaded images but also emojis, avatars in quotes and onebox images. 

#### What's the fix? 

The fix is quite simple: it changes the selector so that avatar and emoji image sizes aren't sent in the payload. This doesn't seem to have any adverse effects, the image sizes are only used as a fallback when limiting the size of images during post processing, and avatars/emojis don't need to be size restricted, they're already small. 

#### Caveats

This isn't a perfect fix. Onebox image will still be considered media, but that is probably desirable behaviour.  Long-term, the heuristic here should not rely on image sizes, but that is a much bigger change, outside the current scope.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/controllers/composer.js [hunk 1]
```diff
@@ -1008,7 +1008,9 @@ export default Controller.extend({
     // TODO: This should not happen in model
     const imageSizes = {};
     document
-      .querySelectorAll("#reply-control .d-editor-preview img")
+      .querySelectorAll(
+        "#reply-control .d-editor-preview img:not(.avatar):not(.emoji)"
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/app/controllers/composer.js:1012]: I think that works as well?

```suggestion
        "#reply-control .d-editor-preview img:not(.avatar, .emoji)"
```

**Review by ZogStriP — APPROVED** [review]

File: app/assets/javascripts/discourse/app/controllers/composer.js (hunk 1 above)
Comment (pmusaraj) [reply app/assets/javascripts/discourse/app/controllers/composer.js:1012]: Naah, that doesn't work, the pseudoclass doesn't take multiple classes: https://css-tricks.com/css-not-with-multiple-classes/.

Comment (ZogStriP) [reply app/assets/javascripts/discourse/app/controllers/composer.js:1012]: That's not what [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/:not#description) says 🤔 


> You can negate several selectors at the same time. Example: `:not(.foo, .bar)` is equivalent to `:not(.foo):not(.bar)`.

Comment (pmusaraj) [reply app/assets/javascripts/discourse/app/controllers/composer.js:1012]: You are right, TIL, this now does work. Initially it didn't, per: https://caniuse.com/css-not-sel-list

**Pull request merged**